code/game/objects/structures/crates_lockers/crates.dm

```diff
@@ -34,6 +34,13 @@
 	if(!src.can_open())
 		return FALSE
 
+	for(var/obj/O in src)
+		if(O.density)
+			var/response = alert(usr, "This crate has been packed with bluespace compression, an item inside won't fit back inside. Are you sure you want to open it?","Bluespace Compression Warning", "No", "Yes")
```

Comment:
This will break things like the knock spell or revenants opening things with defile.
You'll need to solve this at the point where it is sure that a crewmember opens it by hand and not by a power or such

Reply:
I added an argument to the proc to handle that edge case. Let me know if you think of any other cases.